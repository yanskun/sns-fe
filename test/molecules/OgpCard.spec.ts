import { createLocalVue, shallowMount } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import Vuetify from 'vuetify'
import Component from '~/components/molecules/OgpCard.vue'

const localVue = createLocalVue()
localVue.use(VueCompositionApi)
localVue.use(Vuetify)

describe('OgpCard', () => {
  test('snapshot', () => {
    const wrapper = shallowMount(Component, {
      propsData: {
        href: 'https://github.com/',
        src: 'https://github.githubassets.com/images/modules/open_graph/github-logo.png',
        title: 'GitHub',
        description:
          'GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.',
      },
      localVue,
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
