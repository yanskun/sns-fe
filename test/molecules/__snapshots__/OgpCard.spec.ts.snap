// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OgpCard snapshot 1`] = `
<a
  class="ogp-card"
  href="https://github.com/"
  target="_blank"
>
  <v-card-stub
    loaderheight="4"
    tag="div"
  >
    <div
      class="d-flex flex-no-wrap justify-space-between"
    >
      <div
        class="img-wrapper"
      >
        <v-avatar-stub
          class="ma-3"
          size="100"
          tile="true"
        >
          <v-img-stub
            class="ogp-img"
            options="[object Object]"
            position="center center"
            src="https://github.githubassets.com/images/modules/open_graph/github-logo.png"
            transition="fade-transition"
          />
        </v-avatar-stub>
      </div>
      <div>
        <v-card-title-stub>
          GitHub
        </v-card-title-stub>
        <v-card-subtitle-stub>
          GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.
        </v-card-subtitle-stub>
      </div>
    </div>
  </v-card-stub>
</a>
`;
